src/plugins/data/public/search/session/session_service.test.ts

```diff
@@ -156,4 +156,51 @@ describe('Session service', () => {
     expect(sessionService.isCurrentSession('some-other')).toBeFalsy();
     expect(sessionService.isCurrentSession(sessionId)).toBeTruthy();
   });
+
+  test('setupStorage() enables storage capabilities', async () => {
```

Comment:
To imitate someone using this service's api (and not via UI)
Can you add a test for calling the `save` api, when `setupStorage` wasn't called?
It should fail, right?

Also, might be nice to test what happens if `getUrlGeneratorData` throws and error while saving.

Reply:
> Can you add a test for calling the save api, when setupStorage wasn't called?

There is one: 
https://github.com/elastic/kibana/blob/6cd417744d63642dc35903b0fec42ccf0b99f880/src/plugins/data/public/search/session/session_service.test.ts#L162

Or did you mean something else?

Reply:
> Also, might be nice to test what happens if getUrlGeneratorData throws and error while saving.

Added this one